tests/management/integration_test.go

```diff
@@ -29,11 +33,12 @@ func Test_ManagementAPI_Integration(t *testing.T) {
 		t.Run("Create an API request that should pass", func(t *testing.T) {
 			req := service.MyAPICredentialApi.GetApiCredentialDetailsConfig(context.Background())
 
-			resp, httpRes, err := service.MyAPICredentialApi.GetApiCredentialDetails(req)
+			resp, httpRes, serviceErr, err := service.MyAPICredentialApi.GetApiCredentialDetails(req)
```

Comment:
We do not want to add another return value. The 3 we have is already too much.

Reply:
hmm so what do think should be the approach ?
The RestServiceError is standard

Reply:
We can copy it to the `common` package, implement the `error` interface, and return it instead of `common.APIError`. 

In this approach, we can consider just changing `common.SendAPIRequest`, instead of changing templates and all the endpoints.